From: Baolin Wang <baolin.wang@linux.alibaba.com>
Subject: arm64-hugetlb-implement-arm64-specific-hugetlb_mask_last_page-fix
Date: Mon, 27 Jun 2022 13:55:37 +0800

I missed the case that '__PAGETABLE_PMD_FOLDED', was defined, which PUD is
same with PMD.

Link: https://lkml.kernel.org/r/a14e7b39-6a8a-4609-b4a1-84ac574f5c96@linux.alibaba.com
Signed-off-by: Baolin Wang <baolin.wang@linux.alibaba.com>
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/arm64/mm/hugetlbpage.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/arm64/mm/hugetlbpage.c~arm64-hugetlb-implement-arm64-specific-hugetlb_mask_last_page-fix
+++ a/arch/arm64/mm/hugetlbpage.c
@@ -373,8 +373,10 @@ unsigned long hugetlb_mask_last_page(str
 	unsigned long hp_size = huge_page_size(h);
 
 	switch (hp_size) {
+#ifndef __PAGETABLE_PMD_FOLDED
 	case PUD_SIZE:
 		return PGDIR_SIZE - PUD_SIZE;
+#endif
 	case CONT_PMD_SIZE:
 		return PUD_SIZE - CONT_PMD_SIZE;
 	case PMD_SIZE:
_
